game-core/src/main/java/games/strategy/triplea/delegate/RocketsFireHelper.java

```diff
@@ -35,7 +38,60 @@
 /**
  * Logic to fire rockets.
  */
-public class RocketsFireHelper {
+public class RocketsFireHelper implements Serializable {
+  private static final long serialVersionUID = -3694482580987118444L;
+  private final Set<Territory> attackingFromTerritories = new HashSet<>();
+  private final Map<Territory, Territory> attackedTerritories = new LinkedHashMap<>();
+  private final Map<Territory, Unit> attackedUnits = new LinkedHashMap<>();
+  private boolean needToFindRocketTargets;
+
+  private RocketsFireHelper() {}
+
+  /**
+   * Set up rockets for better than V1 rockets.
+   */
+  public static RocketsFireHelper setUpRockets(final IDelegateBridge bridge) {
+    final GameData data = bridge.getData();
+    final RocketsFireHelper object = new RocketsFireHelper();
+    // WW2V2/WW2V3, fires at end of combat move - now moved to the start of the BattleDelegate
+    // WW2V1, fires at end of non combat move so does not call here
+    object.needToFindRocketTargets = false;
+    if (GameStepPropertiesHelper.isFireRockets(data) && TechTracker.hasRocket(bridge.getPlayerId())) {
+      if (Properties.getStrictRockets(data)) {
+        object.needToFindRocketTargets = true;
+      } else {
+        object.findRocketTargets(bridge);
+      }
+    }
+    return object;
+  }
+
+  /**
+   * Fire rockets if we are v1
+   * In this rule set, each player only gets one rocket attack per turn. 
+   */
+  public static void fireWW2V1(final IDelegateBridge bridge) {
+    final GameData data = bridge.getData();
+    if (isWW2V2(data) || isAllRocketsAttack(data)) {
```

Comment:
Just moved it back to MoveDelegate but I think I see the reason. Putting it here has all the rocket code in the one place. I'm inclined to agree that it is awkward here. 

If we do keep it here, we probably should change the method to something like fireWW2V1IfNeeded

Reply:
Do you like the idea of changing the method names or having the check in the move and battle delegates better?

Reply:
If its the same check then probably better to keep it here to avoid duplication and rename the method. If it isn't the same checks the probably better to push it out to where this is called.